app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -138,16 +139,26 @@ class FireDatabase : Database() {
     }
 
     override fun initUserProfile(userModel: UserModel): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+
         val userData = HashMap<String, Any>()
         userData.put(emailFile, userModel.getEmailAddress())
         userData.put(firstnameFile, userModel.getFirstname())
         userData.put(surnameFile, userModel.getSurname())
         userData.put(dateOfBirthFile, userModel.getDateOfBirth().toEpochDay())
-        userData.put(distanceGoalFile, userModel.getDistanceGoal())
-        userData.put(activityTimeGoalFile, userModel.getActivityTime())
-        userData.put(nbOfPathsGoalFile, userModel.getNumberOfPathsGoal())
-
-        return updateUserData(userData)
+        userData.put(currentDistanceGoalFile, userModel.getCurrentDistanceGoal())
+        userData.put(currentActivityTimeGoalFile, userModel.getCurrentActivityTime())
+        userData.put(currentNOfPathsGoalFile, userModel.getCurrentNumberOfPathsGoal())
+
+        updateUserData(userData).thenApply {
+            initUserAchievement().thenAccept { isInit ->
+                if(isInit){
+                    future.complete(true)
+                }
+                future.completeExceptionally(Exception("The user profile was not correctly initiate in the database !"))
```

Comment:
Detail: typo
The user profile was not correctly initiate in the database ! -> The user profile was not correctly initialized in the database !